mods/ra/maps/allies-08a/allies08a-AI.lua

```diff
@@ -0,0 +1,239 @@
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if USSRWarFactory.IsDead or USSRWarFactory.Owner ~= ussr then
+		return
+	end
+
+	ussr.Build({ Utils.Random(SovietVehicles) }, function(units)
+		table.insert(AttackGroup, units[1])
+		SendAttackGroup()
+		Trigger.AfterDelay(ProductionInterval[Map.LobbyOption("difficulty")], ProduceVehicles)
+	end)
+end
+
+ProduceAircraft = function()
+	if (Airfield1.IsDead or Airfield1.Owner ~= ussr) and (Airfield2.IsDead or Airfield2.Owner ~= ussr) then
+		return
+	end
+
+	ussr.Build(SovietAircraftType, function(units)
+		local mig = units[1]
+		Migs[#Migs + 1] = mig
+
+		Trigger.OnKilled(mig, ProduceAircraft)
+
+		local alive = Utils.Where(Migs, function(y) return not y.IsDead end)
+		if #alive < 2 then
+			Trigger.AfterDelay(DateTime.Seconds(ProductionInterval[Map.LobbyOption("difficulty")] / 2), ProduceAircraft)
+		end
+
+		TargetAndAttack(mig)
+	end)
+end
+
+TargetAndAttack = function(mig, target)
+	if not target or target.IsDead or (not target.IsInWorld) then
+		local enemies = Utils.Where(greece.GetActors(), function(actor)
+			return
+				actor.HasProperty("Health") and
+				actor.Type ~= "brik"
+		end)
+		if #enemies > 0 then
+			target = Utils.Random(enemies)
+		end
+	end
+
+	if target and mig.AmmoCount() > 0 and mig.CanTarget(target) then
+		mig.Attack(target)
+	else
+		mig.ReturnToBase()
+	end
+
+	mig.CallFunc(function()
+		TargetAndAttack(mig, target)
+	end)
+end
+
+GroundWaves = function()
+	local path = Utils.Random(GroundAttackPaths)
+	local units = Reinforcements.Reinforce(ussr, Utils.Random(GroundAttackUnits[GroundAttackUnitType]), { path[1] })
+	local lastWaypoint = path[#path]
+	Utils.Do(units, function(unit)
+		Trigger.OnAddedToWorld(unit, function()
+			unit.Patrol(path)
+			Trigger.OnIdle(unit, function()
+				unit.AttackMove(lastWaypoint)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(GroundWavesDelays), GroundWaves)
+end
+
+WTransWaves = function()
+	local way = Utils.Random(WTransWays)
+	local units = Utils.Random(WTransUnits)
+	local attackUnits = Reinforcements.ReinforceWithTransport(ussr, "lst", units , way, { way[2], way[1] })[2]
+	Utils.Do(attackUnits, function(a)
+		Trigger.OnAddedToWorld(a, function()
+			a.AttackMove(AttackChrono.Location)
+			IdleHunt(a)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(WTransDelays), WTransWaves)
+end
+
+Paradrop = function()
+	local units = PowerProxy.SendParatroopers(Utils.Random(ParadropLZs))
+	Utils.Do(units, function(unit)
+		Trigger.OnAddedToWorld(unit, IdleHunt)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(ParadropDelays), Paradrop)
+end
+
+SendParabombs = function()
+	if Airfield1.IsDead or Airfield1.Owner ~= ussr then
+		return
+	end
+
+	local targets = Utils.Where(greece.GetActors(), function(actor)
+		return
+			actor.HasProperty("Sell") and
```

Comment:
What speaks against using `Building` instead of `Sell` here? You could drop the explicit checks for `atek` and `pdox` then.

Reply:
`Building` doesn't work. Won't crash it but the parabombs never come.